### PR TITLE
:bug: Source Bing Ads: fix state setter for bulk streams

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/oncall/issues/3738
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 126, in read
    yield from self._read_stream(
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 203, in _read_stream
    for record in record_iterator:
  File "/usr/local/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 236, in _read_incremental
    stream_instance.state = stream_state  # type: ignore # we check that state in the dir(stream_instance)
  File "/airbyte/integration_code/source_bing_ads/bulk_streams.py", line 63, in state
    current_state_value = self._state.get(str(value["Account Id"]), {}).get(self.cursor_field, "")
KeyError: 'Account Id'
```
If state was provided for reading stream state setter worked with value in the same way as with record. But record notation and state notation are different, so key error was occurred in such cases.

## How
Added checking for 'Account Id' key in received object in state setter to identify how state should be updated.

state notation - 
```
{
{"1287362184"(account id): {"cursor field": "value"},...
}
```
In the created state there is no key 'Account Id' which causes error above when sync reads stream with state (example cmd: ... read ... ---state connection_state.json) when state is provided [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L236). To create state connector uses last record, retrieves Account Id value and cursor value for that account id. But when state have been already created in the previous sync state setter didn't work in the right way and tried to retrieve value for Account Id key which is not exists in the created state. So to handle this issue we check if 'Account Id' exists in the provided to state entity or not.

### CODE REVIEW

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py
Comment (erohmensing) [inline airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py:1]: Is there a unit test we can write to reproduce the old error on the old code and make sure it doesn't occur with the new code?

File: airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml
Comment (erohmensing) [inline airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml:1]: Are these changes related to the code change?

**Review by erohmensing — COMMENTED** [review]
@darynaishchenko I got paged about this issue as it hit a few more workspaces. Stoked to see there's already a fix in the works! Want to help you get it over the line. 

> Added checking for 'Account Id' key in received object in state setter to identify how state should be updated.

I think I understand in general what's going on here (we update state 2 different ways depending on what sort of message is making us update state) - can you elaborate a bit more in the PR description about this difference? Maybe with examples of what the 2 types of messages coming in look like. The unit test I proposed might help with this understanding too!

File: airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/acceptance-test-config.yml:1]: @erohmensing, these changes are added due to dynamic values in expected records to pass CAT.

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py:1]: updated existed unit test

**Comment by darynaishchenko** [comment]
> @darynaishchenko I got paged about this issue as it hit a few more workspaces. Stoked to see there's already a fix in the works! Want to help you get it over the line.
> 
> > Added checking for 'Account Id' key in received object in state setter to identify how state should be updated.
> 
> I think I understand in general what's going on here (we update state 2 different ways depending on what sort of message is making us update state) - can you elaborate a bit more in the PR description about this difference? Maybe with examples of what the 2 types of messages coming in look like. The unit test I proposed might help with this understanding too!

@erohmensing , updated PR description. yes, here is 2 types - record (uses to retrieve needed values and save it to state) and a state. And connector tries to retrieve 'Account Id' key from state, but in state we use account id value to retrieve cursor.

**Review by artem1205 — APPROVED** [review]

**Comment by erohmensing** [comment]
@darynaishchenko great, thanks! I know you used the customer's creds to debug the issue, did you also use the creds to validate that the fix gets them to a green state again?

File: airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py
Comment (erohmensing) [reply airbyte-integrations/connectors/source-bing-ads/source_bing_ads/bulk_streams.py:1]: Really clears it up for me. Thank you :)

**Review by erohmensing — APPROVED** [review]

**Comment by darynaishchenko** [comment]
> @darynaishchenko great, thanks! I know you used the customer's creds to debug the issue, did you also use the creds to validate that the fix gets them to a green state again?

@erohmensing I wasn't able to connect with user creds(get_config script returned outdated creds), but this issue reproduces with our test creds also. So I tested it with our creds.

**Comment by erohmensing** [comment]
@darynaishchenko ok! As long as we've tested it after reproducing and it fixed it, 👍🏻 from me :shipit:
